### PR TITLE
Fix exception with multiselect ChoiceSet, ellipsis with maxLines in TextBlock and truncated Markdown headers

### PR DESCRIPTION
**Pull request by dclaux**
## Related Issue
- Fixes https://github.com/microsoft/AdaptiveCards/issues/5302
- Fixes https://github.com/microsoft/AdaptiveCards/issues/4847
- Fixes https://github.com/microsoft/AdaptiveCards/issues/5249
- Fixes https://github.com/microsoft/AdaptiveCards/issues/5315

## Description
- Code was attempting to call `toString()` on `undefined` when a ChoiceSet input had no selection
- `TextBlock.maxLines` is now handled using the non-standard `--webkit-line-clamp` CSS property. While non-standard, this property is supported across all browsers, except IE. In IE, the renderer will fall back to the old behavior. This change fixes two bugs:
  - Because line clamping takes care of truncation, the renderer doesn't need to specify explicit line heights anymore, which leads to Markdown headings not being cut off anymore
  - Truncated `TextBlock.text` now always has an ellipsis, even if it wraps

## Sample Card
```json
{
    "type": "AdaptiveCard",
    "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
    "version": "1.3",
    "body": [
        {
            "type": "TextBlock",
            "text": "# Lorem ipsum dolor sit amet,\n## consectetur adipiscing elit,\nsed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.",
            "wrap": true,
            "maxLines": 5
        },
        {
            "type": "Input.ChoiceSet",
            "choices": [
                {
                    "title": "Choice 1",
                    "value": "Choice 1"
                },
                {
                    "title": "Choice 2",
                    "value": "Choice 2"
                }
            ],
            "placeholder": "Placeholder text",
            "id": "ds",
            "isMultiSelect": true
        }
    ],
    "actions": [
        {
            "type": "Action.Submit",
            "title": "Action.Submit"
        }
    ]
}
```

## How Verified
Verified manually in adaptivecards-designer-app

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/5307)

### CODE REVIEW

File: source/nodejs/adaptivecards/src/card-elements.ts [hunk 1]
```diff
@@ -1015,8 +1015,20 @@ export class TextBlock extends BaseTextBlock {
...
+                    if (Utils.isInternetExplorer() || !GlobalSettings.useWebkitLineClamp) {
+                        element.style.maxHeight = (this._computedLineHeight * this.maxLines) + "px";
+                    }
+                    else {
+                        // While non standard, --webkit-line-clamp works in every browser (except IE)
+                        // and is a great solution to support the maxLines feature with ellipsis
+                        // truncation. With --webkit-line-clamp there is need to use explicit line heights
+                        element.style.removeProperty("line-height");
+                        element.style.display = "-webkit-box";
+                        element.style.webkitBoxOrient = "vertical";
```
Comment (paulcam206) [inline source/nodejs/adaptivecards/src/card-elements.ts:1029]: looks like box orient [doesn't work by default](https://caniuse.com/?search=-webkit-box-orient) in Firefox? is that something we should be concerned about?

**Review by paulcam206 — APPROVED** [review]
looks good (assuming we're cool with not everything working in Firefox)

File: source/nodejs/adaptivecards/src/card-elements.ts (hunk 1 above)
Comment (dclaux) [reply source/nodejs/adaptivecards/src/card-elements.ts:1029]: I've tested the change in Edge, Chrome, Firefox, Opera and Brave and it worked across the board.

Comment (dclaux) [reply source/nodejs/adaptivecards/src/card-elements.ts:1029]: Ah - and now I remember what happened when I looked at the DOM in Firefox. It actually automatically replaced `--webkit-box-orient` with `-moz-box-orient`:
image

So the question is, how did that happen?
- Is that something new in more recent versions of Firefox?
- Is it TypeScript magic?

I'm going to assume the latter, but I can't find any info.

**Pull request merged**